tpu/src/test/java/tpu/TpuVmIT.java

```diff
@@ -210,4 +210,27 @@ public void testStopTpuVm() throws IOException, ExecutionException, InterruptedE
       assertEquals(returnedNode, mockNode);
     }
   }
+
+  @Test
+  public void testCreateTpuVmWithStartupScript() throws Exception {
+    try (MockedStatic<TpuClient> mockedTpuClient = mockStatic(TpuClient.class)) {
+      Node mockNode = mock(Node.class);
+      TpuClient mockTpuClient = mock(TpuClient.class);
+      OperationFuture mockFuture = mock(OperationFuture.class);
+
+      mockedTpuClient.when(TpuClient::create).thenReturn(mockTpuClient);
+      when(mockTpuClient.createNodeAsync(any(CreateNodeRequest.class)))
+              .thenReturn(mockFuture);
+      when(mockFuture.get()).thenReturn(mockNode);
+
+      Node returnedNode = CreateTpuVmWithStartupScript.createTpuVmWithStartupScript(
+              PROJECT_ID, ZONE, NODE_NAME,
+              TPU_TYPE, TPU_SOFTWARE_VERSION);
+
+      verify(mockTpuClient, times(1))
+              .createNodeAsync(any(CreateNodeRequest.class));
+      verify(mockFuture, times(1)).get();
+      assertEquals(returnedNode, mockNode);
```

Comment:
nit: I am unsure that this assertion is valid. The more secure way is to ensure that values are propagated to the Node object.

Reply:
Thank you! Fixed